### DISCUSSION TITLE
keybinds not intercepting the keys

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#168] [Ideas]
i just got the idea to make a keybind to dismiss notifications to `Escape`.
while this worked, it seems to interfere with the key's regular function.
it would be cool to have keybinds leaving keys' original function intact.

**Comment by YaLTeR** [comment]
How does it work in other compositors, is there a separate binding mode that also passes the key through?

One problem with this is that it will work and feel correct (since most of the time Escape does nothing), but then one day you'll get a notification while working in an app that has something important bound to Escape, leading to accidentally triggering it.

**Reply by KiaraGrouwstra** [reply]
so i'm mostly familiar with hyprland, which offers a couple different [ways to do bindings](https://wiki.hyprland.org/Configuring/Binds/); `bind`, `bindr`, `bindm`, etc.
that being said, over there i had not personally tried to pull off this particular thing, so i'm not sure what the exact equivalent over there would be for this, tho i think they had one.

i agree not intercepting can have side-effects tho, and my use-case here probably isn't the most interesting of the special binds they were handling.

**Reply by YaLTeR** [reply]
I guess this would correspond to `bindn` in Hyprland?

**Reply by KiaraGrouwstra** [reply]
looks like it yeah!
